File: app/src/main/java/com/yuyakaido/android/genesis/presentation/fragment/BaseFragment.java
package com.yuyakaido.android.genesis.presentation.fragment;

import com.trello.rxlifecycle.components.support.RxFragment;
import com.yuyakaido.android.genesis.app.App;

/**
 * Created by yuyakaido on 3/5/16.
 */
public class BaseFragment extends RxFragment {

    @Override
    public void onDestroy() {
        super.onDestroy();
        App.getRefWatcher(getContext()).watch(this);
    }

}

File: cores/presentation/src/main/java/com/yuyakaido/android/gaia/core/presentation/BaseFragment.kt
package com.yuyakaido.android.gaia.core.presentation

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.lifecycle.Lifecycle
import timber.log.Timber

abstract class BaseFragment : Fragment() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    printLog(Lifecycle.Event.ON_CREATE)
  }

  override fun onStart() {
    super.onStart()
    printLog(Lifecycle.Event.ON_START)
  }

  override fun onResume() {
    super.onResume()
    printLog(Lifecycle.Event.ON_RESUME)
  }

  override fun onPause() {
    super.onPause()
    printLog(Lifecycle.Event.ON_PAUSE)
  }

  override fun onStop() {
    super.onStop()
    printLog(Lifecycle.Event.ON_STOP)
  }

  override fun onDestroy() {
    printLog(Lifecycle.Event.ON_DESTROY)
    super.onDestroy()
  }

  private fun printLog(event: Lifecycle.Event) {
    Timber.v("${this::class.java.simpleName}.${event.name} is called.")
  }

}